sources/packages/backend/apps/api/src/route-controllers/education-program/education-program.controller.service.ts

```diff
@@ -128,6 +129,11 @@ export class EducationProgramControllerService {
             new ApiProcessError(error.message, error.name),
           );
         }
+        if (error.name === ENTRANCE_REQUIREMENTS_WRONG_STATE_CODE) {
```

Comment:
We have other errors like `EDUCATION_PROGRAM_NOT_FOUND` and `DUPLICATE_SABC_CODE` at the server side because they are not possible to be validated by the `dryRunSubmission` but this would not be the case for this validation unless I am missing something.
Is there a reason to have this on the server side?

Reply:
yes, sorry. I didn't realize that. Will discard the test I did as we are mocking the dryRun process for now.

Reply:
Don't be sorry 😉 Thanks for doing the changes.